internal/operators/manager.go

```diff
@@ -1,155 +1,281 @@
-	lsoEnabled := false
-	ocsEnabled := mgr.checkOCSEnabled(cluster)
-	if ocsEnabled {
-		lsoEnabled = true // if OCS is enabled, LSO must be enabled by default
-	} else {
-		lsoEnabled = mgr.checkLSOEnabled(cluster)
+	// TODO: cluster should already contain up-to-date list of operators - implemented here for now to replicate
+	// the original behaviour
+	err := mgr.UpdateDependencies(cluster)
+	if err != nil {
+		return nil, err
 	}
+
 	operatorManifests := make(map[string]string)
 
-	if lsoEnabled {
-		manifests, err := mgr.generateLSOManifests(cluster)
-		if err != nil {
-			mgr.log.Error("Cannot generate LSO manifests due to ", err)
-			return nil, err
-		}
-		for k, v := range manifests {
-			operatorManifests[k] = v
-		}
+	clusterOperators, err := unmarshallOperators(cluster.Operators)
+	if err != nil {
+		return nil, err
 	}
-
-	if ocsEnabled {
-		manifests, err := mgr.generateOCSManifests(cluster)
-		if err != nil {
-			mgr.log.Error("Cannot generate OCS manifests due to ", err)
-			return nil, err
-		}
-		for k, v := range manifests {
-			operatorManifests[k] = v
+	// Generate manifests for all the generic operators
+	for _, clusterOperator := range clusterOperators {
+		operator := mgr.operators[clusterOperator.OperatorType]
+		if swag.BoolValue(clusterOperator.Enabled) && operator != nil {
+			manifests, err := operator.GenerateManifests(cluster)
+			if err != nil {
+				mgr.log.Error(fmt.Sprintf("Cannot generate %v manifests due to ", clusterOperator.OperatorType), err)
+				return nil, err
+			}
+			if manifests != nil {
+				for k, v := range manifests.Files {
+					operatorManifests[k] = v
+				}
+			}
 		}
 	}
+
 	return operatorManifests, nil
 }
 
 // AnyOperatorEnabled checks whether any operator has been enabled for the given cluster
 func (mgr *Manager) AnyOperatorEnabled(cluster *common.Cluster) bool {
-	return mgr.checkLSOEnabled(cluster) || mgr.checkOCSEnabled(cluster)
+	for _, operator := range mgr.operators {
+		if isEnabled(cluster, operator.GetType()) {
+			return true
+		}
+	}
+	return false
 }
 
-// ValidateOCSRequirements validates OCS requirements. Returns "true" if OCS operator is not deployed
-func (mgr *Manager) ValidateOCSRequirements(cluster *common.Cluster) string {
-	if isEnabled(cluster, models.OperatorTypeOcs) {
-		return mgr.ocsValidator.ValidateOCSRequirements(&cluster.Cluster)
+// ValidateHost validates host requirements
+func (mgr *Manager) ValidateHost(ctx context.Context, cluster *common.Cluster, host *models.Host) ([]api.ValidationResult, error) {
+	// TODO: cluster should already contain up-to-date list of operators - implemented here for now to replicate
+	// the original behaviour
+	err := mgr.UpdateDependencies(cluster)
+	if err != nil {
+		return nil, err
+	}
+	clusterOperators, err := unmarshallOperators(cluster.Operators)
+	if err != nil {
+		return nil, err
+	}
+
+	results := make([]api.ValidationResult, 0, len(mgr.operators))
+
+	pendingOperators := make(map[models.OperatorType]bool)
```

Comment:
can you explain what is the need of pendingOperators ?

Reply:
It is used to detect which operators were not enabled for which there should be a successful validation information added: https://github.com/openshift/assisted-service/pull/1140/files/df25d249dd081e30b0f8a8a62d993df98f33ab1c#diff-bedca24305045fda0daa5e7b8fee347f6e9c3ec018f6eebddbfe9e55b93fe9d7R125

For example: if user enables only LSO, during cluster or host validation only LSO validation will be executed and we need to add also ValidationResult for not executed OCS validation.

Reply:
please add a remark in the code about it